### PR TITLE
Support for go modules

### PR DESCRIPTION
**Pull request by loafoe**
This PR adds support for building with go modules. The Dockerfile build was also changed so it uses multi-stage build to produce the fhirbase binary. Dependencies are cached in a docker layer.

I have not modified the `Makefile` structure but that would come as a next PR if this is accepted. For now both `Gokpkg.*` and `go.mod` files are there but eventually the `Gopkg` stuff should be removed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,13 +25,6 @@ script:
   - GOOS=windows GOARCH=amd64 BINSUFFIX=-windows-amd64.exe make
   - GOOS=windows GOARCH=386 BINSUFFIX=-windows-386.exe make
   - GOOS=darwin GOARCH=amd64 BINSUFFIX=-darwin-amd64 make
-  - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
-  - "[ ! -z \"$TRAVIS_TAG\" ] && make docker || echo 'non-tag build'"
 
 before_install:
   - curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
-  - curl -L https://github.com/tcnksm/ghr/releases/download/v0.12.0/ghr_v0.12.0_linux_amd64.tar.gz --output ghr.tar.gz && tar zxvf ghr.tar.gz
```
Comment (mlapshin) [inline .travis.yml:33]: Thanks for the PR. Why did you drop building of non-docker binaries and uploading them to GitHub Releases?

Comment (loafoe) [reply .travis.yml:33]: Hi @mlapshin, this step would always fail for someone contributing from the outside no? 
As they would not have permission to push new releases to your project (nor should they ever have).

**Comment by gedw99** [comment]
whats happening with this PR ?
go modules support is almost mandatory these days..

**Comment by mlapshin** [comment]
Thanks @loafoe, it's merged now.

**Pull request merged**